### DISCUSSION TITLE
Sanitization of data through `filter` helper

### DISCUSSION THREAD

**Discussion by Amir-Zouerami** [pocketbase/pocketbase#3791] [Q&A] [answered]
I read in the [SDK docs](https://github.com/pocketbase/js-sdk#binding-filter-parameters) about binding filter parameters and had some questions about it:

- does this sanitize data for me? For instance this example from the docs is safe?
```ts
const record = await pb.collection("example").getList(1, 20, {
  // the same as: "title ~ 'te\\'st' && (totalA = 123 || totalB = 123)"
  filter: pb.filter("title ~ {:title} && (totalA = {:num} || totalB = {:num})", { title: "te'st", num: 123 })
})
```
In the [docs](https://github.com/pocketbase/js-sdk#binding-filter-parameters) it said "on the client-side this is not much of an issue". Why?

- does this work for `create` as well? Let's say for posting comments (`pb.collection('comments').create()`). If so, how would I do it?

- Does this use prepared statements behind the scene? Is there a way to use prepared statements in Pocketbase?

---
_sidenote: any ways to contribute to docs in the future?_

**Comment by ganigeorgiev** [comment, marked as answer]
The JS SDK sends HTTP requests using `fetch`.
The `filter` param is just a query parameter, there are no prepare statements here, just quotes normalization for the string values.

It doesn't matter when it is used in a browser context because the user is in full control of what is being executed and they can send whatever they want.
Generally speaking the same apply when the SDK is used in a JS server-side (node/deno/bun/lambda/etc.) context when the appropriate API rules are used.
**The request is always validated once it reach the PocketBase server based on the related collection API rules.**

An issue in the JS server-side case can arise when for example you use the untrusted user input to fetch some "prefiltered" record or while doing some extra node validation based on the fetch result, for example:

```js
// in some imaginary node/deno/bun/lambda/etc. action ...
function submitHandler({ req }) {
   const slug = req.body["slug"]

   const post = await pb.collection("posts").getFirstListItem(
       // this for example could be unsafe as a malicious user can craft the "slug" string to generate
       // a filter like `slug = "1" || slug = "somethingElse" && status = "active"` to return any other post
       // to prevent it you can use pb.filter(`slug={:slug} && status = "active"`, { slug })
       `slug="${slug}" && status = "active"`
   );

   ...
}
```

> any ways to contribute to docs in the future?

For the site docs, the repo can be found at https://github.com/pocketbase/site.
